modules/apps/transfer/keeper/keeper.go

```diff
@@ -337,6 +337,50 @@ func (k Keeper) deleteForwardedPacket(ctx sdk.Context, portID, channelID string,
 	store.Delete(packetKey)
 }
 
+// getAllForwardedPackets gets all forward packets stored in state.
+func (k Keeper) getAllForwardedPackets(ctx sdk.Context) []types.ForwardedPacket {
+	var packets []types.ForwardedPacket
+	k.iterateForwardedPackets(ctx, func(packet types.ForwardedPacket) bool {
+		packets = append(packets, packet)
+		return false
+	})
+
+	return packets
+}
+
+// iterateForwardedPackets iterates over the forward packets in the store and performs a callback function.
+func (k Keeper) iterateForwardedPackets(ctx sdk.Context, cb func(packet types.ForwardedPacket) bool) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := storetypes.KVStorePrefixIterator(store, types.ForwardedPacketKey)
+
+	defer sdk.LogDeferred(ctx.Logger(), func() error { return iterator.Close() })
+	for ; iterator.Valid(); iterator.Next() {
+		var forwardPacket types.ForwardedPacket
+		k.cdc.MustUnmarshal(iterator.Value(), &forwardPacket.Packet)
+
+		parts := strings.Split(string(iterator.Key()), "/")
+		if len(parts) != 4 {
+			panic(fmt.Errorf("key path should always have 4 elements"))
+		}
```

Comment:
maybe add a comment that the `types.ForwardedPacketKey` is `parts[0]`? A defensive check could be useful as well